src/main/java/com/sonyericsson/jenkins/plugins/bfa/PluginImpl.java

```diff
@@ -576,83 +673,26 @@ public KnowledgeBase getKnowledgeBase() {
-            nrOfScanThreads = DEFAULT_NR_OF_SCAN_THREADS;
-        } else {
-            nrOfScanThreads = scanThreads;
-        }
+    public boolean configure(StaplerRequest req, JSONObject o) {
+        KnowledgeBase existingKb = this.knowledgeBase;
+        req.bindJSON(this, o);
 
-        if (maxLogSize < 0) {
-            maxLogSize = DEFAULT_MAX_LOG_SIZE;
-        }
-
-        if (corePoolNumberOfThreads < ScanOnDemandVariables.DEFAULT_SOD_COREPOOL_THREADS) {
-            sodVariables.setSodCorePoolNumberOfThreads(ScanOnDemandVariables.DEFAULT_SOD_COREPOOL_THREADS);
-        } else {
-            sodVariables.setSodCorePoolNumberOfThreads(corePoolNumberOfThreads);
-        }
-
-        if (jobShutdownTimeWait < ScanOnDemandVariables.DEFAULT_SOD_WAIT_FOR_JOBS_SHUTDOWN_TIMEOUT) {
-            sodVariables.setSodWaitForJobShutdownTimeout(ScanOnDemandVariables.
-                    DEFAULT_SOD_WAIT_FOR_JOBS_SHUTDOWN_TIMEOUT);
-        } else {
-            sodVariables.setSodWaitForJobShutdownTimeout(jobShutdownTimeWait);
-        }
-        if (thrkeepAliveTime < ScanOnDemandVariables.DEFAULT_SOD_THREADS_KEEP_ALIVE_TIME) {
-            sodVariables.setSodThreadKeepAliveTime(ScanOnDemandVariables.DEFAULT_SOD_THREADS_KEEP_ALIVE_TIME);
-        } else {
-            sodVariables.setSodThreadKeepAliveTime(thrkeepAliveTime);
-        }
-        if (minSodWorkerThreads < ScanOnDemandVariables.DEFAULT_MINIMUM_SOD_WORKER_THREADS) {
-            sodVariables.setMinimumSodWorkerThreads(ScanOnDemandVariables.DEFAULT_MINIMUM_SOD_WORKER_THREADS);
-        } else {
-            sodVariables.setMinimumSodWorkerThreads(minSodWorkerThreads);
-        }
-        if (maxSodWorkerThreads < ScanOnDemandVariables.DEFAULT_MAXIMUM_SOD_WORKER_THREADS) {
-            sodVariables.setMaximumSodWorkerThreads(ScanOnDemandVariables.DEFAULT_MAXIMUM_SOD_WORKER_THREADS);
-        } else {
-            sodVariables.setMaximumSodWorkerThreads(maxSodWorkerThreads);
-        }
-        if (maxSodWorkerThreads < ScanOnDemandVariables.DEFAULT_MAXIMUM_SOD_WORKER_THREADS) {
-            sodVariables.setMaximumSodWorkerThreads(ScanOnDemandVariables.DEFAULT_MAXIMUM_SOD_WORKER_THREADS);
-        } else {
-            sodVariables.setMaximumSodWorkerThreads(maxSodWorkerThreads);
-        }
-        KnowledgeBase base = req.bindJSON(KnowledgeBase.class, o.getJSONObject("knowledgeBase"));
-        if (base != null && !knowledgeBase.equals(base)) {
-            try {
-                base.start();
-            } catch (Exception e) {
-                logger.log(Level.SEVERE, "Could not start new knowledge base, reverting ", e);
-                save();
-                return;
-            }
+        if (existingKb != null && !existingKb.equals(knowledgeBase)) {
             if (o.getBoolean("convertOldKb")) {
                 try {
-                    base.convertFrom(knowledgeBase);
+                    knowledgeBase.start();
+                    knowledgeBase.convertFrom(existingKb);
```

Comment:
It doesn't belong there from my understanding of what this does, seems like a hack here anyway. Should be an action from some button I would have thought...

Reply:
The idea from the beginning was that you might start with a local file database, and then later you might want some statistics or you add another Jenkins master and want to share the database. So doing the conversion on the configuration change was sensible in our opinion.
It would be nice if that was supported for JCasC as well, but can be added later I think.

Reply:
I see, can we do it separately please

Reply:
Yes, as I said it can be done at a later date. Doesn't even have to be this year ;)